apps/prairielearn/elements/pl-matrix-component-input/pl-matrix-component-input.py

```diff
@@ -186,15 +189,23 @@ def render(element_html: str, data: pl.QuestionData) -> str:
 
         if parse_error is None:
             a_submitted = pl.from_json(data["submitted_answers"].get(name, None))
-            if a_submitted is not None and len(a_submitted.shape) == 2:
+            if (
+                a_submitted is not None
+                and isinstance(a_submitted, np.ndarray)
+                and len(a_submitted.shape) == 2
+            ):
                 m, n = np.shape(a_submitted)
+            else:
+                raise ValueError(
```

Comment:
Won't this raise an error if a_submitted is None? Isn't that ok?

Reply:
https://github.com/PrairieLearn/PrairieLearn/pull/11188/files/c16f47fdd511d607361168508a27a884a2d02121#diff-a0524518964635a4209ba547ce104822f2bbe50325f46796213320496be0c15cR241

```python
sub_html_table = create_table_for_html_display(
m, n, name, label, data, "output-feedback"
)
```

I had changed this because `create_table_for_html_display` was complaining that `m`/`n` can't be done.

I haven't validated that this branch isn't reachable yet.

Reply:
ahhh ok, just making sure